.github/scripts/simple-smoke-test.sh

```diff
@@ -28,81 +15,39 @@ function remove_container_if_exists() {
     fi
 }
 
-function check_distribution_type() {
-    local image=$1
-    local expected_distribution_type=$2
-
-    echo "Checking ${image} distribution type"
-    if docker run --rm "${image}" bash -c 'compgen -G lib/*enterprise*'; then
-      echo "EE contents identified"
-      distribution_type="ee"
-    else
-      echo "No EE contents identified - assuming OSS"
-      distribution_type="oss"
-    fi
-
-    if [[ "${distribution_type}" != "${expected_distribution_type}" ]]; then
-      echoerr "Image ${image} should contain ${expected_distribution_type} distribution but ${distribution_type} was detected"
-      exit 1
-    fi
-}
-
-function check_image_hz_version() {
-    local image=$1
-    local expected_version=$2
-
-    echo "Checking ${image} version"
-    local version
-    version=$(docker run --rm "${image}" bin/hz-cli --version | awk '/Hazelcast/ {print $2}')
-    if [[ "${version}" == "${expected_version}" ]]; then
-      echo "${image} version identified as ${version}"
-    else
-      echoerr "${image} version was ${version}, not ${expected_version} as expected"
-      exit 1
-    fi
-}
-
-function test_map_read_write() {
-    local image=$1
-    local expected_distribution_type=$2
-
+function start() {
     echo "Starting container '${container_name}' from image '${image}'"
     docker run -it --name "${container_name}" -e HZ_LICENSEKEY -e HZ_INSTANCETRACKING_FILENAME -d -p5701:5701 "${image}"
-    local key="some-key"
-    local expected="some-value"
-    echo "Putting value '${expected}' for key '${key}'"
-    while ! clc --timeout 5s map set -n some-map "${key}" "${expected}" --log.path stderr
-    do
-      echo "Retrying..."
-      sleep 3
-    done
-    echo "Getting value for key '${key}'"
-    local actual
-    actual=$(clc map get --format delimited -n some-map "${key}" --log.path stderr)
-    echo "Stopping container ${container_name}"
-    docker stop "${container_name}"
-
-    if [[ "${expected}" != "${actual}" ]]; then
-        echoerr "Expected to read '${expected}' but got '${actual}'"
-        exit 1;
-    fi
 }
 
-function install_clc() {
-  while ! curl https://hazelcast.com/clc/install.sh | bash
-    do
-      echo "Retrying clc installation..."
-      sleep 3
-    done
-  export PATH=${PATH}:${HOME}/.hazelcast/bin
-  clc config add default cluster.name=dev cluster.address=localhost
+function get_hz_logs() {
+    docker logs "${container_name}"
 }
 
-# Prints the given message to stderr
-function echoerr() {
-  echo "ERROR - $*" 1>&2;
+function stop() {
+    echo "Stopping container ${container_name}"
+    docker stop "${container_name}"
 }
 
-
-install_clc
-test_docker_image "$@"
+image=$1
+container_name=$2
+input_distribution_type=$3
+expected_version=$4
+
+case "${input_distribution_type}" in
```

Comment:
nit - can we move this to a function()?
Looks little naked

Reply:
e676ec1f3a41b47de0fcd9a4f8b9503d1fa911bf